upstream-community-operators/mongodb-enterprise/mongodb.v0.9.0.clusterserviceversion.yaml

```diff
@@ -0,0 +1,325 @@
+      kind: MongoDbShardedCluster
+      version: v1
+      resources:
+      # FIXME(alecmerdler): These are probably supposed to be different resources for the different CRD configurations
+      - kind: Service
+        name: ''
+        version: v1
+      - kind: StatefulSet
+        name: ''
+        version: v1beta2
+      - kind: Pod
+        name: ''
+        version: v1
+      specDescriptors:
+      - description: Credentials for Ops Manager or Cloud Manager.
+        displayName: Credentials
+        path: credentials
+        x-descriptors:
+        - 'urn:alm:descriptor:io.kubernetes:core:v1:Secret'
+
+      - description: Project this deployment belongs to.
+        displayName: Project
+        path: project
+        x-descriptors:
+        - 'urn:alm:descriptor:io.kubernetes:core:v1:ConfigMap'
+
+      - description: MongoDB version to be installed.
+        displayName: Version
+        path: version
+        x-descriptors:
+        - 'urn:alm:descriptor:com.tectonic.ui:label'
+
+      - description: How many Config Servers will be deployed
+        displayName: Config Server Count
+        path: configServerCount
+        x-descriptors:
+        - 'urn:alm:descriptor:com.tectonic.ui:label'
+
+      - description: How many MongoDB Servers per Shard will be deployed
+        displayName: MongoDB Servers per Shard
+        path: mongodsPerShardCount
+        x-descriptors:
+        - 'urn:alm:descriptor:com.tectonic.ui:label'
+
+      - description: How many mongos will be deployed
+        displayName: Mongos (MongoDB Shard) to be deployed
+        path: mongosCount
+        x-descriptors:
+        - 'urn:alm:descriptor:com.tectonic.ui:label'
+
+      - description: Amount of Shards to be deployed
+        displayName: Shards
+        path: shardCount
+        x-descriptors:
+        - 'urn:alm:descriptor:com.tectonic.ui:label'
+
+      - description: The service where this MongoDB deployment will be listening.
+        displayName: Service
+        path: service
+        x-descriptors:
+        - 'urn:alm:descriptor:io.kubernetes:Service'
+  installModes:
+  - type: OwnNamespace
+    supported: true
+  - type: SingleNamespace
+    supported: true
+  - type: MultiNamespace
+    supported: false
+  - type: AllNamespaces
+    supported: true
+  install:
+    strategy: deployment
+    spec:
+      deployments:
+      - name: mongodb-enterprise-operator
+        spec:
+          replicas: 1
+          selector:
+            matchLabels:
+              k8s-app: mongodb-enterprise-operator
+          template:
+            metadata:
+              labels:
+                k8s-app: mongodb-enterprise-operator
+            spec:
+              serviceAccountName: mongodb-enterprise-operator
+              imagePullSecrets:
+                - name: ''
+
+              containers:
+              - name: mongodb-enterprise-operator
+                image: registry.connect.redhat.com/mongodb/enterprise-operator:0.9.1
```

Comment:
Does this image require authentication to pull? I believe the upstream version should use a publicly available image; if no such image is available then the upstream changes can be reverted

Reply:
What do you recommend in this case? Should I use quay.io for these images?

Reply:
The upstream-community-operators version of Mongodb v0.3.2 uses a quay.io image: https://github.com/operator-framework/community-operators/blob/master/upstream-community-operators/mongodb-enterprise/mongodb.v0.3.2.clusterserviceversion.yaml#L10

Reply:
Ok, I didn't know about this, as this CSV file was originally meant to be included in OLM and Openshift, which I understood had access to registry.connect.

I'll push the images to Quay and update the file.

Reply:
This CSV file is the same one that's meant to be installed with OLM if I'm not mistaken. It means that people using this on Openshift will have to use our non-RHEL7 images. Is that correct?

Reply:
People on OpenShift will be given the `community-operators` CSV for this operator. `upstream-community-operators` are for vanilla Kubernetes, which should have a non-registry image, I beleive. @dmesser please correct me if I'm wrong

Reply:
I set the images in "upstream" directory to the ones publicly available in quay.io.